fuel-indexer/indexer/src/executor.rs

```diff
@@ -98,31 +115,39 @@ impl IndexExecutor {
     }
 
     /// Trigger a WASM event handler, passing in a serialized event struct.
-    pub fn trigger_event(&self, event_name: &str, bytes: Vec<u8>) -> IndexerResult<()> {
-        let alloc_fn = self
-            .instance
-            .exports
-            .get_native_function::<u32, u32>("alloc_fn")?;
-        let memory = self.instance.exports.get_memory("memory")?;
+    pub fn trigger_event(&self, event_name: &str, bytes: Vec<Vec<u8>>) -> IndexerResult<()> {
+        let mut args = Vec::with_capacity(bytes.len());
+        for arg in bytes.into_iter() {
+            args.push(ffi::WasmArg::new(&self.instance, arg)?)
+        }
+        let arg_list = ffi::WasmArgList::new(&self.instance, args.iter().collect())?;
 
         if let Some(handlers) = self.events.get(event_name) {
             for handler in handlers.iter() {
                 let fun = self
                     .instance
                     .exports
-                    .get_native_function::<(u32, u32), ()>(handler)?;
+                    .get_native_function::<(u32, u32, u32), ()>(handler)?;
 
-                let wasm_mem = alloc_fn.call(bytes.len() as u32)?;
-                let range = wasm_mem as usize..wasm_mem as usize + bytes.len();
-
-                unsafe {
-                    // Safety: the alloc call for wasm_mem has succeeded for bytes.len()
-                    //         so we have this block of memory for copying. The fun.call() below
-                    //         will release it.
-                    memory.data_unchecked_mut()[range].copy_from_slice(&bytes);
+                if let Ok(db) = self.db.lock() {
+                    db.start_transaction()?;
+                } else {
+                    return Err(IndexerError::LockPoisoned);
```

Comment:
Given that a poisoned lock is likely an unrecoverable error and each subsequent request to this database lock for the lifetime of this program would fail, this is one of the few scenarios where I think an `.expect()` is appropriate vs an error result.